core/src/main/java/org/mskcc/cbio/portal/scripts/UpdateGenePanel.java

```diff
@@ -0,0 +1,145 @@
+        Set<CanonicalGene> canonicalGenes = extractGenes(properties, true);
+        GenePanel genePanel = DaoGenePanel.getGenePanelByStableId(stableId);
+
+        if (genePanel == null) {
+            ProgressMonitor.logWarning("Gene panel " + stableId + " does not exist in the database! Exiting.");
+            return;
+        }
+
+        System.err.println("WARNING: You are about to make the following changes to gene panel " + stableId + ":");
+        DaoGenePanel.updatePreview(genePanel, canonicalGenes);
+        System.err.println("Proceed? This action is irreversible. (Y/n)");
+
+        if (promptUser()) {
+            if (DaoSampleProfile.sampleProfileMappingExistsByPanel(genePanel.getInternalId())) {
+                ProgressMonitor.logWarning("Gene panel " + stableId
+                    + " already exists in database and is being used. Proceeding with gene panel update!");
+            } else {
+                ProgressMonitor.logWarning("Gene panel " + stableId
+                    + " already exists in database but is unused. Proceeding with gene panel update!");
+            }
+            DaoGenePanel.updateGenePanel(genePanel, canonicalGenes);
+        }
+    }
+
+    private static boolean promptUser() {
+        boolean confirmed = false;
+        String answer = s.nextLine().toUpperCase();
+
+        switch (answer) {
+            case "Y":
+                System.err.println("Proceeding with update script.");
+                confirmed = true;
+                break;
+            case "N":
+                System.err.println("Exiting update script.");
+                break;
+            default:
+                System.err.println("Unrecognized input, exiting.");
+                break;
+        }
+
+        return confirmed;
+    }
+
+    public void setFile(File genePanelFile) {
+        this.genePanelFile = genePanelFile;
+    }
+
+    /**
+     * Makes an instance to run with the given command line arguments.
+     *
+     * @param args the command line arguments to be used
+     */
+    public UpdateGenePanel(String[] args) {
+        super(args);
+    }
+
+    /**
+     * Runs the command as a script and exits with an appropriate exit code.
+     *
+     * @param args the arguments given on the command line
+     */
+    public static void main(String[] args) {
+        ConsoleRunnable runner = new UpdateGenePanel(args);
+        System.err.println(
+            "WARNING: This script is ONLY for updating a gene panel, not for adding a new gene panel. To add a new gene panel, run the ImportGenePanel script.");
+        System.err.println("If you are making changes to an existing gene panel due to a miscount of genes, proceed.");
+        System.err.println(
+            "Otherwise, if you would like to add a new version of a gene panel, please exit this script and import the gene panel with a new stable ID!");
+        System.err.println("Will you proceed to update an existing gene panel? (Y/n)");
```

Comment:
This is super picky, but I don't like the multiple method calls here. Let's just make this one big block. Also, I added a few more new lines. I like to stick to about 80 columns for console output. Oh, also, why is this going to standard err instead of standard out? This isn't an error message.
```suggestion
        System.out.print(
            "WARNING: This script is ONLY for updating a gene panel, not for adding a new gene panel.\n" + 
            "To add a new gene panel, run the ImportGenePanel script.\n" +
            "If you are making changes to an existing gene panel due to a miscount of genes, proceed.\n" +
            "Otherwise, if you would like to add a new version of a gene panel, please exit this script\n" +
            "and import the gene panel with a new stable ID!\n" +
            "Will you proceed to update an existing gene panel? (Y/n)\n"
        );
```

Reply:
I reckoned that the red text from the System.err.print looks more alerting than the black text from System.out.print, but yes it is not an error message perse